spinoso-symbol/src/casecmp/unicode.rs

```diff
@@ -0,0 +1,63 @@
+//! Unicode case folding comparisons for byte content resolved from `Symbol`s.
+
+use artichoke_core::intern::Intern;
+use core::str;
+
+use focaccia::CaseFold;
+
+/// Compare the byte contents of two symbols using Unicode case-folding
+/// comparison for equality.
+///
+/// The byteslice associated with each symbol is resolved via the given
+/// interner. Unresolved symbols are compared as if they resolve to `&[]`.
+///
+/// This comparison function attempts to convert each symbol's byte content to a
+/// UTF-8 [`str`]. If both symbols resolve to UTF-8 contents, [Unicode case
+/// folding] is used when comparing the contents and this function returns
+/// `Ok(Some(bool))`. If neither symbol resolves to UTF-8 contents, this
+/// function falls back to [`ascii_casecmp`] and returns `Ok(Some(bool))`.
+/// Otherwise, the two symbols have byte contents with different encodings and
+/// `Ok(None)` is returned.
+///
+/// This function can be used to implement [`Symbol#casecmp?`] for the
+/// [`Symbol`] type defined in Ruby Core.
+///
+/// # Errors
+///
+/// If the interner returns an error while retrieving a symbol, that error is
+/// returned. See [`Intern::lookup_symbol`].
+///
+/// [Unicode case folding]: https://www.w3.org/International/wiki/Case_folding
+/// [`ascii_casecmp`]: crate::casecmp::ascii_casecmp
+/// [`Symbol#casecmp?`]: https://ruby-doc.org/core-2.6.3/Symbol.html#method-i-casecmp-3F
+/// [`Symbol`]: https://ruby-doc.org/core-2.6.3/Symbol.html
+#[inline]
+#[cfg_attr(docsrs, doc(cfg(feature = "artichoke")))]
+pub fn case_eq<T, U>(
```

Comment:
this function extracts bytes from the interner, does encoding detection and delegates to `focaccia`.

This function only supports `Encoding::UTF_8` and `Encoding::BINARY`.